mitodl/open-discussions
# Search index for staff lists

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes https://github.com/mitodl/open-discussions/issues/3870

#### What's this PR do?
Add public staff lists to the search index
Hides user list search behind a feature flag

#### How should this be manually tested?
Create a mix of public and private staff lists with at least 1 item each, as described in this PR.
With your browser's network tab open, go to http://od.odl.local:8063/learn/search?type=stafflist
Find the call in the network tab to http://localhost:8063/learn/search?type=stafflist, and check the response.  Your public staff lists should be included in the `hits` section of the JSON.


Search for user lists (create some first if necessary), you should get 0 results back: ttp://localhost:8063/learn/search?type=userlist
Set `FEATURE_USER_LIST_SEARCH=True` in your .env file, restart containers, then search again.  Now you should see your userlists.

Run `manage.py recreate_index --stafflists`, then repeat the search for staff lists above, the same results should be returned.
